### ISSUE TITLE
Is there a way to load test data in docker?

### ISSUE THREAD

**Issue by CyborTronik** [goauthentik/authentik#5700]
Context:
I write an app (experiment with Spring boot service). For security I need an auth service. Ideally is to reuse an existing solution like Authentik. During the development phase I do write integration and system tests. Which means I need to run the auth service during those tests. For that a docker-compose was configured which looks to be fine for now.

Question:
Can I configure (or automate) authentik, so during docker start it would load few test users and an app credentials that I can use in tests?

Thanks

**Comment by BeryJu** [comment]
You can do that with blueprints, see https://goauthentik.io/developer-docs/blueprints/

for example the blueprint I use for some testing users:

```yaml
# yaml-language-server: $schema=https://goauthentik.io/blueprints/schema.json
version: 1
metadata:
  name: local - dev users & groups
  labels:
    blueprints.goauthentik.io/instantiate: "false"
entries:
  - model: authentik_core.group
    identifiers:
      name: Department foo
    id: group_foo
  - model: authentik_core.group
    identifiers:
      name: Department bar
    id: group_bar
  - model: authentik_core.group
    identifiers:
      name: Department baz
    id: group_baz
  - model: authentik_core.user
    identifiers:
      attributes:
        managed: "user1"
    attrs:
      username: user1
      name: user1
      groups:
        - !KeyOf group_foo
  - model: authentik_core.user
    identifiers:
      attributes:
        managed: "user2"
    attrs:
      username: user2
      name: user2
      groups:
        - !KeyOf group_bar
        - !KeyOf group_foo
  - model: authentik_core.user
    identifiers:
      attributes:
        managed: "user3"
    attrs:
      username: user3
      name: user3
      groups:
        - !KeyOf group_baz
        - !KeyOf group_bar

```

**Comment by CyborTronik** [comment]
Thank you @BeryJu , Looks good, however is not clear to set the passwords - which seems to be a blocker

**Comment by CyborTronik** [comment]
Use case scenario context below:
I do want to write an app, and the idea is to focus on the solution:

- As many auth services are available on the market, only few offers local docker support (which is essential if you want to use best practices). That's why authentik has a good spot
- For automation tests there should be some data loaded into it: users (with passwords) and roles
- Another essential precondition is to have the application(client) registered properly: also automated data (app id and secret)
- A really nice to have (or maybe even crucial if the target audience is developers) is to have an article (example) on how to use it with an web service (backend only could be enough). Doesn't matter the stack ,a popular one best for SEO.

**Comment by CyborTronik** [comment]
It feels like something has been fixed but not released yet

**Comment by BeryJu** [comment]
The PR above adds support to set user's passwords via blueprints, and will be released with 2023.6 (and also usable with the current beta version)
